app/src/main/java/android/example/com/split/data/entity/Group.java

```diff
@@ -24,7 +24,7 @@
     /**
      * Members ({@link User}s) in the group
      */
-    private List<String> members;
```

Comment:
@blagicaz  This changes breaks the back-end code, the GroupDataRepository saves groups in the back-end (Firebase Firestore) assuming that members are referenced in a group by their IDs, and retrieves them later based on that. Changes the members list from String to User will make all the back-end logic related to group members get broken.

@SalehIbra I think it would be a good idea to add tests for the public methods in user and data repositories, and then whenever anyone has updates they get to run these and other tests, this will tell if I am making changes that break other parts of the code not directly affecting the feature I am working on.

Reply:
@blagicaz You can leave the new utility methods you added if you need them, just annotate them using `@com.google.firebase.firestore.Exclude`so that Firestore ignores them while writing and reading from the back-end. And return all the fields and methods you removed/modified. If renaming is required, rename the new fields and methods not the old ones.

Reply:
I made the needed changes and approved the pull-request.